Automattic/sensei
# Fix conflicts with modules using the same name

Fixes #5777

### Changes proposed in this Pull Request

* Use the lastTitle to sync the course structure with the editor blocks only when the block is a module.


### More context
Currently, when the module name is updated, a new module (with a new id) is created on the backend, so on the editor, we lose the ability to sync the course structure with the outline blocks, so is necessary to use the fields title and lastTitle to sync them, ignore the id.

Before this change, we were applying the same strategy to sync the lessons and it was messing with the editor when two different lessons were using the same title, causing bugs and browser crashing. 



### Testing instructions

**Test 1:** 
* Go to the branch
* Try to reproduce the steps described in issue #5777
* Check if they are not reproducible. 

**Test 2**
* Create 2 modules
* Apply a background color on the first one
* Save the course
* Reload the page
* Check if the background color is applied (it means the sync happened with success)
